### PR TITLE
 Use != to support international number formats

### LINKED ISSUE #1808: gravity.sh invalid operator syntax error (plural detection)
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

4

---
**Actual behavior**

/opt/pihole/gravity.sh shows following error (German):

> /opt/pihole/gravity.sh: Zeile 455: [[: 44.559: Syntaxfehler: Ungültiger arithmetischer Operator. (Fehlerverursachendes Zeichen ist \".559\").

Line 455 has following content:
`plural=""; [[ "${num}" -ne 1 ]] && plural="s"`

The problem is that $num has thousand separators, which cannot be parsed by "-ne".
Actually, $num is "44.559"  (fullstop is the German thousand separator)

**Fix:**

Find (2 times):
`plural=; [[ "${num}" -ne 1 ]] && plural="s"`

Replace (2 times):
`plural=; [ "${num}" != "1" ] && plural="s"`

### PR DESCRIPTION
**Pull request by Mcat12**
**By submitting this pull request, I confirm the following:**
- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [x] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#git-commit---signoff))

---

**What does this PR aim to accomplish?:**

Fix #1808

**How does this PR accomplish the above?:**

Use the != operator instead of -ne when comparing numbers, such as when we determine plural output.

**What documentation changes (if any) are needed to support this PR?:**

None

### CODE REVIEW

File: gravity.sh [hunk 1]
```diff
@@ -461,7 +461,7 @@ gravity_Whitelist() {
   fi
 
   num=$(wc -l < "${whitelistFile}")
```
Comment (dracenmarx) [inline gravity.sh]: Please change to

`num=$(printf "%'.0f" "$(wc -l < "${whitelistFile}")")`

File: gravity.sh [hunk 2]
```diff
@@ -114,7 +114,7 @@ gravity_DNSLookup() {
   echo -ne "  ${INFO} Waiting up to ${secs} seconds before continuing..."
   until timeout 1 getent hosts "${lookupDomain}" &> /dev/null; do
     [[ "${secs:-}" -eq 0 ]] && break
-    [[ "${secs:-}" -ne 1 ]] && plural="s"
+    [[ "${secs:-}" != 1 ]] && f="s"
```
Comment (dracenmarx) [inline gravity.sh]: You have accidently changed "plural" to "f"

File: gravity.sh [hunk 3]
```diff
@@ -114,7 +114,7 @@ gravity_DNSLookup() {
   echo -ne "  ${INFO} Waiting up to ${secs} seconds before continuing..."
   until timeout 1 getent hosts "${lookupDomain}" &> /dev/null; do
     [[ "${secs:-}" -eq 0 ]] && break
```
Comment (dracenmarx) [inline gravity.sh]: Please change
`-eq 0`
to
`== 0`

**Comment by jacobsalmela** [comment]
@dracenmarx I implemented some of your requested changes as mentioned here: https://github.com/pi-hole/pi-hole/pull/1831.  

Will you test this branch out with the latest changes?

File: gravity.sh [hunk 4]
```diff
@@ -460,9 +460,9 @@ gravity_Whitelist() {
     return 0
   fi
 
-  num=$(wc -l < "${whitelistFile}")
-  [[ "${num}" -ne 1 ]] && plural="s"
-  str="Whitelisting ${num} domain${plural}"
+  num=$(printf "%'.0f" "$(wc -l < "${whitelistFile}")")
```
Comment (dracenmarx) [inline gravity.sh]: This line should be
`num=$(wc -l < "${whitelistFile}")`

File: gravity.sh [hunk 5]
```diff
@@ -440,7 +440,7 @@ gravity_WhitelistBLD() {
 
   # Create array of unique $sourceDomains
   mapfile -t uniqDomains <<< "$(awk '{ if(!a[$1]++) { print $1 } }' <<< "$(printf '%s\n' "${sourceDomains[@]}")")"
-  [[ "${#uniqDomains[@]}" -ne 1 ]] && plural="s"
+  [[ "${#uniqDomains[@]}" != 1 ]] && plural="s"
 
   str="Adding ${#uniqDomains[@]} blocklist source domain${plural} to the whitelist"
```
Comment (dracenmarx) [inline gravity.sh]: This line could be
`str="Adding $(printf "%'.0f" "${#uniqDomains[@]}") blocklist source domain${plural} to the whitelist"`

**Comment by jacobsalmela** [comment]
We are looking at implementing this in place of this PR: #1850

**Comment by Mcat12** [comment]
Closing in favor of #1850

**Pull request closed without merging**